src/exec/builtins/functions.rs

```diff
@@ -134,42 +149,85 @@ pub fn list_directory(shell: &mut ShellState, args: Vec<&str>) -> Result<()> {
 
     directories.append(&mut files);
 
-    let permission_view = match octal_permissions {
-        true => DirectoryListPermissionMode::Octal,
-        false => DirectoryListPermissionMode::String,
+    let permission_view = { 
+        if hide_permissions {
+            DirectoryListPermissionMode::Hidden
+        } else {
+            match octal_permissions {
+                true => DirectoryListPermissionMode::Octal,
```

Comment:
Probably also should be a `ValueEnum` here, you can have it set to `String` by default, but this one is up to you.